`examples/syntax-test.lus` updates

lorchrob opened the pull request:
Updates to `examples/syntax-test.lus`.

1. Make sure all the syntax is up to date (e.g., using `'(a, b)` rather than `{a, b}` for tuple literals)
2. Remove features we don't support (e.g. various structural assignments and array slices)
3. Add in the new features
    * polymorphism 
    * arbitrary-width machine integers 
    * unbounded subranges
    * frame blocks 
    * if blocks 
    * sets and maps
    * history type constructor
    * refinement types
    * `any` operator
    * `transparent` and `opaque` modifiers

Let me know if I forgot anything 🙂

lorchrob committed:
Update examples/syntax-test.lus to not use deprecated syntax or unsupported features

lorchrob committed:
Add frame and if blocks to syntax-test.lus

lorchrob committed:
Add sets, maps, history to syntax-test.lus

lorchrob committed:
Add refinement types to syntax-test.lus

lorchrob committed:
Minor

daniel-larraz commented:
The new `syntax-test.lus` file triggers a `LustreExpr.Type_mismatch` with the current version on `develop`. Could you please take a look?

In addition, I am missing an example of the [nondeterministic choice operator](https://kind.cs.uiowa.edu/kind2_user_doc/2_input/1_lustre.html#nondeterministic-choice-operator) `any` and the use of the `transparent`/`opaque` [modifiers](https://kind.cs.uiowa.edu/kind2_user_doc/1_techniques/1_techniques.html#modifiers-to-control-node-function-abstraction).

lorchrob committed:
Add any, transparent, and opaque to syntax-test.lus

lorchrob commented:
Previous commit addresses the comment about `any`, `transparent`, and `opaque`. 

I will take a look at the crash (I suppose the fix should be in a separate PR).

lorchrob committed:
Workaround for syntax-test.lus for kind 2 shadowing bugs
